src/System.IO/tests/BufferedStream/BufferedStream.FakeStream.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace System.IO.Tests
+{
+    /// <summary>
+    /// Provides a simple method of testing around a Stream with specific read/write/seek privileges.
+    /// </summary>
+    public class FakeStream : Stream
+    {
+        MemoryStream wrapped;
+        bool canWrite;
+        bool canRead;
+        bool canSeek;
```

Comment:
Nit: these should all be explicitly marked as private, should have underscore prefixes, etc.